src/Database/PostgreSQL/PQTypes/Checks.hs

```diff
@@ -399,6 +402,10 @@ checkDBStructure options tables = fmap mconcat . forM tables $ \(table, version)
       runQuery_ $ sqlSelect "pg_catalog.pg_attribute a" $ do
         sqlResult "a.attname::text"
         sqlResult "pg_catalog.format_type(a.atttypid, a.atttypmod)"
+        sqlResult . parenthesize . toSQLCommand $
+          sqlSelect "pg_catalog.pg_collation c, pg_catalog.pg_type t" $ do
+            sqlResult "CAST(c.collname as text)"
+            sqlWhere "c.oid = a.attcollation AND t.oid = a.atttypid AND a.attcollation <> t.typcollation"
```

Comment:
Why is the last condition needed? To not fetch the default collation? If so, a comment would be nice.

Reply:
If you remove the condition then presumably the checks will fail because all `text` columns will have the default collation which is not in the column definition. 

Can you check that's the case?

Reply:
Changing:
```haskell
            sqlWhere "c.oid = a.attcollation AND t.oid = a.atttypid AND a.attcollation <> t.typcollation"
```
to:
```haskell
            sqlWhere "c.oid = a.attcollation AND t.oid = a.atttypid"
```
results in all the tests still passing, if that was what you meant?

Reply:
Yeah. Interesting :thinking: But let's not experiment too much, original `psql` code should work just fine.